### PR TITLE
Refactor/separate auth initialisation

### PR DESCRIPTION
**Pull request by alexanderleegs**
This PR sets up the scaffolding for https://github.com/isomerpages/isomercms-backend/issues/375. Previously, we had a lot of duplicated routes, due to the need to specify each authenticated route again inside our auth middleware router. This PR modifies the auth middleware such that it is initialised in a separate `newmiddleware/index` file, and each individual router imports the middleware to utilise its `verifyJwt` or `whoami` methods. 

As the auth middleware is no longer a router which is always hit before the actual routes, the check for unknown routes has been moved out to `server.js`. In addition, the `noVerify` method has been removed as it is no longer useful, given that routes which do not need verification can now simply be used without middleware - I am open to adding this back if it provides us with greater clarity that those specific routes do not need verification.

### CODE REVIEW

File: newmiddleware/auth.js [hunk 1]
```diff
@@ -31,320 +26,7 @@ class AuthMiddleware {
...
-    }
-    auth.verifiedPost = (path) => {
-      auth.post(path, this.verifyJwt)
-    }
-
-    // Health check
-    auth.get("/v2/ping", this.noVerify)
-
-    // V1 endpoints
-    // TODO [#351]: Remove after refactor is done
```
Comment (seaerchin) [inline newmiddleware/auth.js:53]: is #351 still relevant?

File: routes/index.js [hunk 2]
```diff
@@ -1,19 +0,0 @@
-const express = require("express")
-const uuid = require("uuid/v4")
-
-const { CLIENT_ID } = process.env
-const { REDIRECT_URI } = process.env
-
-const router = express.Router()
-
-/* GET home page. */
-function getCmsHomepage(req, res) {
```
Comment (seaerchin) [inline routes/index.js:10]: could i check why this is deleted?

**Review by seaerchin — APPROVED** [review]
lgtm (i didn't actually check to see that every route in the removed `getRouter` method has the `verifyJwt` method attached but it should be?)

we should also create a path alias for `@newmiddleware` to ease the imports (but i'm ok w/ not changing the existing `@root/newmiddleware/*`

File: routes/index.js (hunk 2 above)
Comment (seaerchin) [reply routes/index.js:10]: just saw that it's unused - how are we getting the index now then

File: newmiddleware/auth.js (hunk 1 above)
Comment (alexanderleegs) [reply newmiddleware/auth.js:53]: Yup, will remove the V1 endpoints after all routes have been migrated

File: routes/index.js (hunk 2 above)
Comment (seaerchin) [reply routes/index.js:10]: resolved over slack - the above is correct

**Pull request merged**